core/src/provider.rs

```diff
@@ -182,44 +180,6 @@ fn is_known_zk_evm_compatible_chain(chain: Chain) -> Option<bool> {
 }
 
 impl JsonRpcCachedProvider {
-    pub async fn new(
```

Comment:
logic was spread across two separate functions
- `JsonRpcCachedProvider` method
- `create_provider` function

don't see a need for that as we can nicely abstract all in single function. `JsonRpcCachedProvider` has private fields that are only accessible in this module so there is also no other way to create the struct than by calling `create_provider` method

On top of that having `Struct::new()` is not fully idiomatic rust given constructors should be not async.